pkg/cri/resource-manager/cache/affinity.go

```diff
@@ -178,11 +164,88 @@ func (e *Expression) Evaluate(container Container) bool {
 }
 
 // KeyValue extracts the value of the expresssion key from a container.
-func (e *Expression) KeyValue(container Container) (string, bool) {
-	value, ok := container.GetLabel(e.Key)
+func (e *Expression) KeyValue(c Container) (string, bool) {
+	value, ok, _ := c.(*container).resolveRef(e.Key)
 	return value, ok
 }
 
+// resolveRef walks an object trying to resolve a reference to a value.
+func (c *container) resolveRef(path string) (string, bool, error) {
+	var obj interface{}
+
+	c.cache.Debug("resolving %s/%s...", c.PrettyName(), path)
+
+	obj = c
+	ref := strings.Split(path, "/")
+	if len(ref) == 1 {
+		ref = []string{"labels", path}
+	}
+	for {
```

Comment:
Would `for len(ref) > 0` work?

Reply:
I'm not sure I understand... The idea here is that we default to looking up labels if only a key is provided without any `key-domain/`-prefix.

Reply:
We have this
```
		if len(ref) == 0 {
			break
		}
```
later in the loop so it looks like a while loop to me

Reply:
Ah, that one... Yes, you're right, it is essentially a while loop. I'll update it.